rfcs/rdm-0064-custom-fields.md

```diff
@@ -0,0 +1,588 @@
+
+### Overriding how fields will be displayed in the record's / community's detail page
+
+The custom fields will be displayed by default in a _to-be-defined_ place in the record's landing page and community's detail page. The default place will be decided to fit the up-to-date UX design of these pages. That said, every institution has their own UI design and it is already possible to override the default display by overriding the Jinja templates of these pages.
+
+### Adding a custom field as facet in the search page
+
+Search filtering or facetting is configured as it was before, using the and `RDM_FACETS` variable. The `CFTermsFacet` facet class would abstract all the namespacing. Therefore, only the field name is needed (i.e. experiment).
+
+```python
+RDM_FACETS = {  # To add search filtering
+    "experiment": CFTermsFacet()
+}
+```
+
+Then the facet will be displayed in the search page:
+
+![](./0064/facets.png)
+
+## Example
+
+The full configuration for the example use case presented at the beggining is:
+
+_At CERN, I want to choose the experiment to which the record belong to, assuming experiments come from a controlled vocabulary. In addition, I want to be able to filter the record search by experiment._
+
+```python
+RDM_CUSTOM_FIELDS = {
+    "experiment": VocabularyCF(
+        name="experiment",
+        vocabulary_id="cernexperiments",
+    ),
+}
+
+RDM_CUSTOM_FIELDS_UI = [{
+    "section": _("CERN Fields"),
+    "fields": [{
+        "field": "experiment",  # this should be validated against the defined fields in `RDM_CUSTOM_FIELDS`
+        "ui_widget": "MyCustomField",
+        "props": {
+            "label": _('CERN Experiment'),
+            "placeholder": "Type an experiment...",
+            "icon": "pencil",
+            "description": _(
+                "You should fill this field with one of the experiments e.g LHC, ATLAS etc.",
+            )
+        },
+    }]
+}]
+
+RDM_FACETS = {  # To add search filtering
+    "experiment": CFTermsFacet()
+}
+```
+
+Most of the configuration above is self describing. The first variable (`RDM_CUSTOM_FIELDS`) defines which fields are available in the record data model, while the second (`RDM_CUSTOM_FIELDS_UI`) defines which ones are displayed and how (allowing to "hide" some of them).
+
+Before being able to use the field, it has to be created. This command only needs to be run in the first time before its usage.
+
+```shell
+invenio-cli custom-fields create -f experiments
+```
+
+Now, the field is fully available in the records data model under `custom_fields.experiments` and the users will see it in the upload form.
+
+## How we teach this
+
+This feature will be documented in the `customization` section of the InvenioRDM documentation. It should include:
+
+- How to configure the custom fields in the `invenio.cfg` file.
```

Comment:
Question: What would the Table of Contents look like ?

Reply:
Hmmm, I think it would be very similar to the actual points/enummeration of this section. Although not all fit there (e.g. CLI part should be briefly talked about in the docs and the point to the "reference guide".

Reply:
discussion: ok